firmware/Core/Src/telecommands/telecommand_executor.c

```diff
@@ -17,6 +17,17 @@
 uint32_t TCMD_total_tcmd_queued_count = 0;
 uint64_t TCMD_latest_received_tcmd_timestamp_sent = 0;
 
+
+//the head of the circular buffer I think this should work
+
+//make these globals AND static.
+
+
+///@brief  The head of the circular buffer of timestamps of telecommands that have been sent.
+uint8_t  TCMD_timestamp_sent_head = 0;
+/// @brief The circular buffer of timestamps of telecommands that have been sent.
+uint64_t TCMD_timestamp_sent_buffer[TCMD_TIMESTAMP_RECORD_SIZE] = {0};
```

Comment:
Consider renaming from "buffer" to "store" or "record". 

Buffers are typically temporary stores.